apis/apiextensions/v1/composition_environment.go

```diff
@@ -239,10 +239,23 @@ type EnvironmentSourceSelectorLabelMatcher struct {
 	// ValueFromFieldPath specifies the field path to look for the label value.
 	ValueFromFieldPath *string `json:"valueFromFieldPath,omitempty"`
 
+	// FromFieldPathPolicy specifies how to patch from a field path. The default is
+	// 'Required', which means the patch should fail if the path specified via valueFromFieldPath does not exist.
+	// Use 'Ignore' if instead you want it to result in a no-op.
+	// +kubebuilder:validation:Enum=Optional;Required
+	// +optional
+	FromFieldPathPolicy *FromFieldPathPolicy `json:"policy,omitempty"`
```

Comment:
```suggestion
	FromFieldPathPolicy *FromFieldPathPolicy `json:"fromFieldPathPolicy,omitempty"`
```

Should we make it more obvious?

Otherwise, this looks like below in the API, right?

```yaml
          matchLabels:
            - type: FromCompositeFieldPath
              key: stage
              valueFromFieldPath: spec.someValue
              policy: Optional
```

This looks like matching this label (i.e. `matchLabels[0]`) is `Optional`, which I believe is kind of true if `spec.someValue` is not there, we ignore this label. However, it could be confusing in the sense that, even that value is there, this label is optional, which doesn't make sense though. For example, things get interesting if this is used in conjunction with `value` instead:

```yaml
        matchLabels:
          - type: Value
            key: stage
            value: prod
            policy: Optional
```